### PR TITLE
techdebt: [PL-22518]: Adding proper description and mandatory propert…

### PR DESCRIPTION
**Pull request by boopesh007**
…y to SM Connector fields.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30961)

Had been tested on a PR Environment, Automation has been run on the same.

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger messagecheck

**Comment by boopesh007** [comment]
trigger ti0

**Comment by boopesh007** [comment]
trigger ti1

**Comment by boopesh007** [comment]
trigger ti2

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awskmsconnector/AwsKmsConnectorDTO.java [hunk 1]
```diff
@@ -50,15 +50,20 @@
...
   @SecretReference
   @ApiModelProperty(dataType = "string")
   @NotNull
   SecretRefData kmsArn;
-  @Schema(description = "Region for AWS KMS.") private String region;
+
+  @NotNull @Schema(description = SecretManagerDescriptionConstants.AWS_REGION_KMS) private String region;
   @Schema(description = SecretManagerDescriptionConstants.DEFAULT) private boolean isDefault;
-  @JsonIgnore private boolean harnessManaged;
+  @Schema(description = SecretManagerDescriptionConstants.HARNESS_MANAGED) @JsonIgnore private boolean harnessManaged;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awskmsconnector/AwsKmsConnectorDTO.java:66]: This is ignored in JSON, hence we don't need open api description for this filed.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java [hunk 2]
```diff
@@ -74,7 +77,7 @@
   @SecretReference
   @ApiModelProperty(dataType = "string")
   private SecretRefData secretId;
-  private boolean isDefault;
+  @Schema private boolean isDefault;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:80]: why empty Schema?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java [hunk 3]
```diff
@@ -115,9 +118,27 @@ public void validate() {
       throw new InvalidRequestException(
           String.format("Invalid value for secret engine version: %s", secretEngineVersion), INVALID_REQUEST, USER);
     }
+
+    if (getAccessType() == AccessType.APP_ROLE) {
+      if (isBlank(appRoleId)) {
+        throw new InvalidRequestException(
+            "You must provide a App Role Id if you are using AppRole Authentication for Vault.", INVALID_REQUEST, USER);
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:125]: We will need @rashminandasahoo review for the error messages.

File: 120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java [hunk 4]
```diff
@@ -134,18 +139,24 @@ public void testCreateSecretManagerConnectorShouldFail_exceptionFromManager() th
...
+                                    .connector(ConnectorInfoDTO.builder()
+                                                   .connectorType(ConnectorType.VAULT)
+                                                   .connectorConfig(VaultConnectorDTO.builder()
+                                                                        .isDefault(false)
+                                                                        .authToken(secretRefData)
+                                                                        .vaultUrl(randomAlphabetic(10))
+                                                                        .namespace(randomAlphabetic(5))
+                                                                        .build())
+                                                   .build())
+                                    .build()));
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java:159]: Can we have a test case for testing the mandatory fields i.e. If mandatory fields are not passed function should throw an error.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java [hunk 5]
```diff
@@ -74,7 +77,7 @@
   @SecretReference
   @ApiModelProperty(dataType = "string")
   private SecretRefData secretId;
-  private boolean isDefault;
+  @Schema private boolean isDefault;
   @Schema(description = SecretManagerDescriptionConstants.SECRET_ENGINE_VERSION) private int secretEngineVersion;
   @Schema(description = SecretManagerDescriptionConstants.DELEGATE_SELECTORS) private Set<String> delegateSelectors;
   @Schema(description = SecretManagerDescriptionConstants.NAMESPACE) private String namespace;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:83]: If namespace is not mandatory (acc to current behaviour), should we also update UI screen to show this field as optional?

**Comment by meenakshi-harness** [comment]
@boopesh007 can you please run the api automation for these changes in PR env or update in PR description if already tested. Thanks

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awskmsconnector/AwsKmsConnectorDTO.java (hunk 1 above)
Comment (boopesh007) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awskmsconnector/AwsKmsConnectorDTO.java:66]: ok

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java (hunk 5 above)
Comment (boopesh007) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:83]: I already did this, I will update the ticket

Comment (boopesh007) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:83]: will inform in the standup tomorrow and get a UI ticket created.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java (hunk 3 above)
Comment (boopesh007) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:125]: ok will get it reviewed from her tomorrow

**Comment by boopesh007** [comment]
trigger ti4

File: 120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java [hunk 6]
```diff
@@ -134,22 +140,42 @@ public void testCreateSecretManagerConnectorShouldFail_exceptionFromManager() th
...
+  @Category(UnitTests.class)
+  public void testthrowExceptionIfMandatoryFieldsNotPassed() {
+    SecretManagerConfigDTO secretManagerConfigDTO = random(VaultConfigDTO.class);
+    when(defaultConnectorService.get(any(), any(), any(), any())).thenReturn(Optional.empty());
+    when(ngSecretManagerService.createSecretManager(any())).thenReturn(secretManagerConfigDTO);
+    when(defaultConnectorService.create(any(), any())).thenReturn(null);
+    when(connectorRepository.updateMultiple(any(), any())).thenReturn(null);
+    ConnectorDTO requestDTO = getRequestDTO();
+    ((VaultConnectorDTO) requestDTO.getConnectorInfo().getConnectorConfig()).setVaultUrl(null);
+    secretManagerConnectorService.create(requestDTO, ACCOUNT);
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java:176]: Did not get what are we asserting here in the test case.

Comment (boopesh007) [reply 120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java:176]: @meenakshi-harness we are asserting if the test case throws invalidRequest Exception

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 5 of 7 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30961)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Comment by shashank-harness** [comment]
___
*[954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awssecretmanager/AwsSecretManagerDTO.java, line 90 at r4](https://reviewable.io/reviews/harness/harness-core/30961#-MxoH_EA8g_aQcjOQYNO:-MxoH_EA8g_aQcjOQYNP:blbc27m) ([raw file](https://github.com/harness/harness-core/blob/c966ba942a76f4b377252302740d96a1ade97040/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awssecretmanager/AwsSecretManagerDTO.java#L90)):*
> ```Java
>     AwsSecretManagerCredentialType credentialType = this.credential.getCredentialType();
>     Preconditions.checkNotNull(region, "Region cannot be empty");
>     if (MANUAL_CONFIG.equals(credentialType)) {
> ```

Please check how validate method is written in AwsKmsConnectorDTO and do it in similar way. Some validations are missing like roleArn is manadatory for STS_ROLE

**Review by shashank-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30961)* status: 8 file(s) to review.

**Review by shashank-harness — APPROVED** [review]
:lgtm:

> Reviewed 5 of 7 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30961#-:-MxoJC-80gsCj6I0aSNy:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**